zerver/views/message_fetch.py

```diff
@@ -945,13 +986,17 @@ def get_messages_backend(request: HttpRequest, user_profile: UserProfile,
                 # debugged the case that makes it happen.
                 raise Exception(str(err), message_id, narrow)
 
+    allow_edit_history = False
+    if not is_web_public_query:
+        allow_edit_history = user_profile.realm.allow_edit_history
```

Comment:
This bit and many similar shouldn't need to be refactored; we should have an unambiguous `realm` object.

Reply:
Done